garden_ai/app/garden.py

```diff
@@ -177,61 +177,62 @@ def create(
     local_data.put_local_garden(garden)
 
     if verbose:
-        metadata = local_data.get_local_garden(garden.uuid)
+        metadata = json.dumps(local_data.get_local_garden_by_uuid(garden.uuid))
         rich.print_json(metadata)
     return
 
 
-# TODO: allow selecting by DOI.
 @garden_app.command(no_args_is_help=True)
 def add_pipeline(
-    garden_uuid: str = typer.Option(
+    garden_id: str = typer.Option(
         ...,
         "-g",
         "--garden",
-        prompt="Please enter a the name or DOI of a garden",
+        prompt="Please enter the UUID or DOI of a garden",
         help="The name of the garden you want to add a pipeline to",
         rich_help_panel="Required",
     ),
-    pipeline_uuid: str = typer.Option(
+    pipeline_id: str = typer.Option(
         ...,
         "-p",
         "--pipeline",
-        prompt="Please enter a the title or DOI of a pipeline",
+        prompt="Please enter a the UUID or DOI of a pipeline",
         help="The name of the pipeline you want to add",
         rich_help_panel="Required",
     ),
 ):
-    garden_metadata = get_garden_meta(garden_uuid)
-    pipeline_uuids_already_present = set(
-        pipeline["uuid"] for pipeline in garden_metadata["pipelines"]
-    )
-    if pipeline_uuid in pipeline_uuids_already_present:
-        logger.info(f"Pipeline {pipeline_uuid} is already in Garden {garden_uuid}")
+    garden_metadata = get_garden_meta(garden_id)
+    pipeline_ids_already_present = set()
+    for pipeline in garden_metadata["pipelines"]:
```

Comment:
🎄🌟 Nitpick-o-rama! 🌟🎄 Instead of that 🔄 loop-de-loop, accumulating 📚 IDs in a 🧩 set, and then doing a 🔍 membership check outside the loop, why not just ⚡️ zap ⚡️ it in there with this ✨ magical ✨ condition: `if pipeline_id in {pipeline['doi'], pipeline['uuid']}`? 🧙‍♂️🎁

Reply:
```suggestion
    for pipeline in garden_metadata["pipelines"]:
        if pipeline_id in {pipeline['doi'], pipeline['uuid']}:
            logger.info(f"Pipeline {pipeline_id} is already in Garden {garden_id}")
            return
```

Reply:
I dig it